docs/source/models.rst

```diff
@@ -358,22 +294,162 @@ and the output is the batch size and is thus set to -1 to allow for variable bat
         outputs: '[{"name": "output", "type": "string"}]'
-        params: '[{"name": "temperature", "type": "float", "default": 0.5, "shape": null},
-                  {"name": "top_k", "type": "integer", "default": 1, "shape": null},
-                  {"name": "suppress_tokens", "type": "integer", "default": [101, 102], "shape": [-1]}]'
+        params: '[
+            {
+                "name": "temperature", 
+                "type": "float", 
+                "default": 0.5, 
+                "shape": null
+            },
+            {
+                "name": "suppress_tokens", 
+                "type": "integer", 
+                "default": [101, 102],
+                 "shape": [-1]
+            }
+        ]'
+
+The values for inference params are passed to the model at inference time in the form of a dictionary 
+and each param value will be validated against the corresponding param type defined in the model signature. 
+The following example demonstrates how to define params in a model signature and how to use them for model inference.
+
+.. code-block:: python
+
+    import mlflow
+    from mlflow.models import infer_signature
+
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, ctx, model_input, params):
+            return list(params.values())
+
+
+    params = {"temperature": 0.5, "suppress_tokens": [101, 102]}
+    # params' default values are saved with ModelSignature
+    signature = infer_signature(["input"], params=params)
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            python_model=MyModel(), artifact_path="my_model", signature=signature
+        )
+
+    loaded_model = mlflow.pyfunc.load_model(model_info.model_uri)
+
+    # Not passing params -- predict with default values
+    loaded_predict = loaded_model.predict(["input"])
+    assert loaded_predict == [0.5, [101, 102]]
+
+    # Passing some params -- add default values
+    loaded_predict = loaded_model.predict(["input"], params={"temperature": 0.1})
+    assert loaded_predict == [0.1, [101, 102]]
+
+    # Passing all params -- override
+    loaded_predict = loaded_model.predict(
+        ["input"], params={"temperature": 0.5, "suppress_tokens": [103]}
+    )
+    assert loaded_predict == [0.5, [103]]
+
+.. note::
+    When validating param values, the values will be converted to python native types.
+    For example, ``np.float32(0.1)`` will be converted to ``float(0.1)``.
+
+.. _supported-data-types:
+
+Supported Data Types in Model Signature
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Currently, :py:class:`MLflow DataType <mlflow.types.DataType>` defines following data primitives for model signature.
+
+* string
+* integer
+* boolean
+* double
+* float
+* long
+* datetime
+
+Model signature also supports composite data types as follows,
+
+* Array
+* Object (a.k.a. dictionary in python)
+* Nested Arrays and Objects e.g. ``Array(Object(Propertiy("a", Array(DataType.integer))))``
```

Comment:
We might not need to explicitly list this here as it's also Array. We can explain in Array section about what internal types we support.